src/libraries/System.Runtime.Extensions/tests/System/Random.cs

```diff
@@ -7,34 +7,92 @@ namespace System.Tests
 {
     public static class RandomTests
```

Comment:
I think it would be worth having some tests to sanity test the correct bounds on the various Next() overloads , eg
```
        [Theory]
        [InlineData(0, 1)]
        [InlineData(0, 49)]
        [InlineData(1, 5)]
        [InlineData(5, 48)] // I would have a few more, or even use seeded Random
        public static void CheckBounds_Next(int lower, int upper)
        {
            var r = new Random();
            while (true) {
                 int next = r.Next(lower, upper);
                 if (next == lower) break;
                Assert.True(next >= lower && next < upper);
            }
            while (r.Next(lower, upper) != upper - 1) { }
        }
```
and the same for the other overloads, including the single parameter ones, all with narrow ranges of course!

These would complete very quickly and catch an off by one.

Reply:
How is that substantially different from https://github.com/dotnet/runtime/pull/47085/files#diff-622f9060c212af0efe3d764cdbc282bc8ee71bce8125a31438ddee3344ea1d50R46 ? I think I'm missing what's missing.

Reply:
They don't verify that both ends of the range are reached. Say you had a bug that you implemented Next(int a, int b) as if it was a request for values in the range [a, b-1) or [a+1,b) instead of [a,b).. A simple off by one. Those tests would still pass.